subsys/spm/Kconfig

```diff
@@ -71,6 +71,16 @@ config SPM_SERVICE_READ
 
 endif
 
+config SPM_ALLOW_NON_SECURE_RESET
+	bool "Allow Non-Secure reset"
+	default y
```

Comment:
Assuming that we agree to do like this (i.e. allow by default a Non-Secure call to NVIC_SystemReset()), which I really think we do not need to, since there are lots' of options to address this, e.g. workaround only for Thingy, or only for boards without PIN Reset capability, etc....

...we may need to fix the alignment with SPM_SERVICE_REBOOT. These two, can't be default y at the same time (unless I am missing something here). That is, if we don't lock the Non-Secure system reset, what is the point of setting the secure service to reboot?

Reply:
SPM_SERVICE_REBOOT does not interfere with the ability of doing NVIC_SystemReset() so they can both be enabled at the same time.

The point of having secure service reboot with this enabled would be if you have written code which uses the secure service but you want to use a debugger to debug your code. This means you want to have the API exposed so that your code can use it but  you also want to leave it open so that the debugger can reset the device. At least that's a use-case I can see.

Reply:
SPM_ALLOW_NON_SECURE_RESET is not a debugger-specific option. Neither the name, nor the actual functionality suggests it.

Setting SPM_ALLOW_NON_SECURE_RESET allows the NS firmware to directly call NVIC_SystemReset() in the code. So we can't be having a parallel Secure Service that instructs the Secure firmware to call NVIC_SystemReset(). We need to sort this out because it will be confusing, and, in fact, not needed. It will consume flash, ram and execution resources, for no reason.

Have you looked at the definition of SPM_SERVICE_REBOOT?

```
config SPM_SERVICE_REBOOT
	bool "Request a reboot"
	default y
	select REBOOT
	help
	  The Non-Secure Firmware is not allowed to reboot the chip. This
	  service allows it to request a reboot from the SPM.
```

So, it claims that if the option is enabled, the non-secure FW can't issue a system reset, which is clearly, wrong.

Kconfig options are there, so we and product developers may use them to configure our software. We can't be having contradicting options enabled.

Reply:
@ioannisg :
> SPM_ALLOW_NON_SECURE_RESET is not a debugger-specific option. Neither the name, nor the actual functionality suggests it.

Because it's not, I was merely stating a use-case not the pure rationale behind the option which has been discussed in the JIRA issue and this PR. It has to do with devices which do not support pin-reset etc. Also, the options include the following help text, which is also the functionality which it coincidentally toggles on and off.

```
If not enabled, a debugger will not be able to issue a soft reset while the
core is executing Non-Secure code.
```

> Have you looked at the definition of SPM_SERVICE_REBOOT?

Since this PR has been bouncing back and forth on the issue I abstained from updating the help text for other options, and I would be careful to do so before we can agree on what should be implemented.

> So, it claims that if the option is enabled, the non-secure FW can't issue a system reset, which is clearly, wrong.

The statement is wrong as you said and the configuration would be ambiguous if both allow non-secure reset and `SPM_SERVICE_REBOOT` are enabled which of course is never a good thing, nice spot.

I do, however, wonder what the best scheme of having this configurable would be any suggestions? One option is to change the text of the option you listed to:

```
If the Non-Secure Firmware is not allowed to reboot the chip.
This service will allow it to request a reboot from the SPM.
```

This way, the configurations would not be contradicting. Of course, this does not address the issue by having both enabled you'll consume a few more bytes of flash. By default, I do agree both should not be enabled, but I don't think they should be coupled where the rationale has been mentioned before here:

> I am not entirely sure using the existing CONFIG_SPM_SERVICE_REBOOT is the best idea. If this PR is to be used as a solution to the problem of not being able to program the Thingy91, the develpoer would have to choose between
> A) being able to program with current tools and
> B) Using the spm_request_system_reboot in their application

Our current predicament is that we can't have secure and ease of use chose both. In this way, I like @pizi-nordic suggestion of having this "HARDENED MODE" where product developers who are focused on security can turn it on. However we should be careful of claiming too much also that would be out-of-scope for this PR.

I also think both support @Rallare and maintainers @ioannisg should look revisit this commit and see if it can be reverted, when a workaround for the issue with regards to reset has been resolved in tooling.

As for now, I see the argument of having something working with an opened attack vector is stronger than something that does not work for product developer as long as it is communicated how and why they should close the attack vector when releasing their product.